subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/generator/FileContentGenerator.groovy

```diff
@@ -100,13 +100,15 @@ abstract class FileContentGenerator {
             if (!isRoot) {
                 return null
             }
-            if (config.subProjects == 0) {
+            if (config.subProjects == 0 && config.featurePreviews.length == 0) {
                 return ""
             }
             """ 
             ${(0..config.subProjects - 1).collect { "include(\"project$it\")" }.join("\n")}
```

Comment:
This will probably fail if there are no sub-projects, right? Or does this work?

Reply:
It would produce `project-1` for single-project build. Corrected.